pkg/tfbridge/provider.go

```diff
@@ -416,6 +417,25 @@ func (p *Provider) Configure(ctx context.Context,
 	return &pulumirpc.ConfigureResponse{}, nil
 }
 
+var requiredFieldRegex = regexp.MustCompile("\"(.*?)\": required field is not set")
+
+func (p *Provider) formatFailureReason(res Resource, reason string) string {
+	// If a required field is missing and the value can be set via config,
+	// extend the error with a hint to set the proper config value
+	name := requiredFieldRegex.FindStringSubmatch(reason)
+	if len(name) == 2 {
+		field := res.Schema.Fields[name[1]]
+		if field != nil && field.Default != nil {
+			if configKey := field.Default.Config; configKey != "" {
```

Comment:
Very nice to see this can be handled specifically for `Config` defaults - so that this narrowly targets the cases where we introduce this issue in the first place!